src/node/BrowserFetcher.ts

```diff
@@ -572,23 +573,32 @@ function httpRequest(
 
   const proxyURL = getProxyForUrl(url);
   if (proxyURL) {
-    if (url.startsWith('http:')) {
-      const proxy = URL.parse(proxyURL);
-      options = {
-        path: options.href,
-        host: proxy.hostname,
-        port: proxy.port,
-      };
-    } else {
-      const parsedProxyURL = URL.parse(proxyURL);
-
-      const proxyOptions = {
-        ...parsedProxyURL,
-        secureProxy: parsedProxyURL.protocol === 'https:',
-      } as ProxyAgent.HttpsProxyAgentOptions;
-
-      options.agent = new ProxyAgent(proxyOptions);
-      options.rejectUnauthorized = false;
+    const parsedProxyURL = URL.parse(proxyURL);
+    const proxyOptions = {
+      ...parsedProxyURL,
+      secureProxy: false,
+    };
+    switch (true) {
```

Comment:
IMHO not to best way handling those (may be anti-pattern too)

Idk if TS had such thing:
```
let options = when {
  proxyURL.match(...) -> new SocksProxyAgent(...)
  url.startsWith(...) -> new HttpsProxyAgent(...)
}
```

Reply:
switch/case could be used for that but I think going with a classic if/elseif is the better way here to keep the code simple.